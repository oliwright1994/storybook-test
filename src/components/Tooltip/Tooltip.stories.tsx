import Text from '@bright/components/Text'
import createStory from '@bright/storybook/createStory'
import { Box, Icon } from '@chakra-ui/core'
import { boolean, radios, text } from '@storybook/addon-knobs'
import React from 'react'

import Tooltip from './Tooltip'

createStory(
  'COMPONENTS | Tooltip',
  [
    {
      label: 'Tooltip',
      jsx: () => {
        const tooltip = text('Tooltip', 'This is the tooltip text')
        const ariaLabel = text('Aria Label', 'This is the aria-label')
        const tooltipAnchor = radios('Tooltip Anchor', { Text: 'Text', Icon: 'Icon' }, 'Icon')
        const shouldWrapChildren = boolean('Should Wrap Children (in <span>)', true)
        const tooltipAnchorJSX = {
          Text: 'Text Anchor',
          Icon: <Icon color="gray.500" name="question-outline" />,
        }
        return (
          <Box p="l">
            <Text>
              Hover or focus for tooltip: &nbsp;
              <Tooltip label={tooltip} aria-label={ariaLabel} shouldWrapChildren={shouldWrapChildren}>
                {tooltipAnchorJSX[tooltipAnchor]}
              </Tooltip>
            </Text>
          </Box>
        )
      },
    },
  ],
  {
    liveEdit: {
      Component: Tooltip,
      render: 'return <Tooltip label="Tooltip" aria-label="Tooltip" > Edit this example Tooltip. </Tooltip>',
    },
  }
)
