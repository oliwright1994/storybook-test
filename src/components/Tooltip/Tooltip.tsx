import { Tooltip as ChakraTooltip, TooltipProps as IChakraTooltip } from '@chakra-ui/core'
import React, { FC } from 'react'

export type ITooltip = Pick<IChakraTooltip, 'label' | 'aria-label' | 'shouldWrapChildren'>

const Tooltip: FC<ITooltip> = ({ children, label, shouldWrapChildren = true, 'aria-label': ariaLabel }) => {
  return (
    <ChakraTooltip
      hasArrow
      placement="top"
      shouldWrapChildren={shouldWrapChildren}
      label={label}
      aria-label={ariaLabel}
    >
      {children}
    </ChakraTooltip>
  )
}

export default Tooltip
