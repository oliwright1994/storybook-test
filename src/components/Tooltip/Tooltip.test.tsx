import { act, fireEvent, screen } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Tooltip from '.'

beforeAll(() => {
  jest.useFakeTimers()
})

afterAll(() => {
  jest.useRealTimers()
})

describe('<Tooltip />', () => {
  const anchorText = 'Test Text'

  const defaultProps = {
    label: 'Tooltip label',
    'aria-label': 'Tooltip aria-label',
    children: anchorText,
  }

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(<Tooltip {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('should render passed children correctly', () => {
    const { getByText } = renderWithBrightTheme(<Tooltip {...defaultProps} />)

    expect(getByText(anchorText)).toBeTruthy()
  })

  it('should display tooltip on hover', () => {
    const { getByText } = renderWithBrightTheme(<Tooltip {...defaultProps} />)

    const tooltipAnchor = getByText(anchorText)
    expect(screen.queryByRole('tooltip')).toBeNull()
    act(() => {
      fireEvent.mouseOver(tooltipAnchor)
      jest.advanceTimersByTime(200)
    })
    expect(screen.queryByRole('tooltip')).toBeTruthy()

    act(() => {
      fireEvent.mouseOut(tooltipAnchor)
      jest.advanceTimersByTime(200)
    })
    expect(screen.queryByRole('tooltip')).toBeNull()
  })

  it('should display tooltip on focus', () => {
    const { getByText } = renderWithBrightTheme(<Tooltip {...defaultProps} />)

    const tooltipAnchor = getByText(anchorText)
    expect(screen.queryByRole('tooltip')).toBeNull()
    act(() => {
      fireEvent.focus(tooltipAnchor)
      jest.advanceTimersByTime(200)
    })
    expect(screen.queryByRole('tooltip')).toBeTruthy()

    act(() => {
      fireEvent.blur(tooltipAnchor)
      jest.advanceTimersByTime(200)
    })
    expect(screen.queryByRole('tooltip')).toBeNull()
  })

  it('should render aria-label correctly', () => {
    const { getByText } = renderWithBrightTheme(<Tooltip {...defaultProps} />)

    const tooltipAnchor = getByText(anchorText)
    act(() => {
      fireEvent.mouseOver(tooltipAnchor)
      jest.advanceTimersByTime(200)
    })
    expect(screen.getByText(defaultProps['aria-label'])).toBeTruthy()
  })
})
