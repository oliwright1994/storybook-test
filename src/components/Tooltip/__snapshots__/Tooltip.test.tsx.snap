// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render as expected 1`] = `
<div>
  <span
    class="css-0"
    tabindex="0"
  >
    Test Text
  </span>
</div>
`;
